types/index.d.ts

```diff
@@ -2,7 +2,7 @@ import { ClientMetrics, ConsumerGlobalConfig, ConsumerTopicConfig, GlobalConfig,
 
 export type ConsumeActionFunction = (err: LibrdKafkaError, messages: Message[]) => void;
 
-export type ListenActionFunction = (arg: Message) => void;
+export type ListenActionFunction = (arg: Message, err?: Error) => void;
```

Comment:
We utilise node-rdkafka's event actions for `listen` functionality and `data` event in node-rdkafka expect callback with single argument (msg). So, we cannot really change the function signature here. We can rather avoid passing error to callback and just log it maybe in wrapper itself?

Reply:
That seems plausible. Either way no point in sending the error to the client since they wouldn't have any use for it.
Logging it & moving on to make sure the consumer isn't blocked seems best, will make the changes 👍🏻